File: Go/Linked List/83_Remove Duplicates from Sorted List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	cur := head
	if head == nil {
		return nil
	}
	next := head.Next
	for next != nil {
		if next.Val != cur.Val {
			cur, next = cur.Next, next.Next
		} else {
			cur.Next = next.Next
			next = next.Next
		}
	}
	return head
} 

File: C++/Linked List/83_Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution
{
public:
    ListNode *deleteDuplicates(ListNode *head)
    {
        if (head == NULL)
            return head;
        ListNode *first = head;
        ListNode *second = head;
        while (second != NULL)
        {
            while (second != NULL && first->val == second->val)
            {
                second = second->next;
            }
            first->next = second;
            first = second;
            // second = second->next;
        }
        return head;
    }
};